subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java

```diff
@@ -52,7 +54,13 @@ protected CodeQualityExtension createExtension() {
         extension = project.getExtensions().create("checkstyle", CheckstyleExtension.class, project);
         extension.setToolVersion(DEFAULT_CHECKSTYLE_VERSION);
 
-        extension.setConfigDir(project.file("config/checkstyle"));
+        if (usesSubprojectCheckstyleConfiguration()) {
+            DeprecationLogger.nagUserOfDeprecated("Setting the Checkstyle configuration file under 'config/checkstyle' of a sub project", "Use the root project's 'config/checkstyle' directory instead");
```

Comment:
❌ This warning will be emitted eagerly even if the user really wants a checkstyle file specific to this subproject. We should use a lazy approach instead.

Reply:
Can you elaborate how a lazy approach would prevent us from rendering the message? Maybe this speaks for not rendering a deprecation message at all as we want to allow user to set the location explicitly via `configDir` and don't really deprecate the sub project's location. The only thing that changed was the default location.